### PR TITLE
En_Skjneedle OK

### PR DESCRIPTION
**Pull request by Half-asian**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 1]
```diff
@@ -14,8 +14,10 @@ void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Draw(Actor* thisx, GlobalContext* globalCtx);
+int func_80B01F6C(Actor* thisx);
+
+extern Gfx* D_06000EB0[];
```
Comment (Random06457) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:19]:
```suggestion
extern Gfx D_06000EB0[];
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 2]
```diff
@@ -14,8 +14,10 @@ void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Draw(Actor* thisx, GlobalContext* globalCtx);
+int func_80B01F6C(Actor* thisx);
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:17]:
```suggestion
s32 func_80B01F6C(EnSkjneedle* this);
```
Does this function need a generic actor argument? It seems to be specific to only this actor.

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 3]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
+	{ 10, 4, -2, { 0, 0, 0 } },
+};
+
+static InitChainEntry initChain[] = {
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
+
+void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = (EnSkjneedle*)thisx;
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:47]:
```suggestion
	EnSkjneedle* this = THIS;
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 4]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
+	thisx->flags &= ~0x1;
+	Actor_SetScale(&this->actor, 0.01f);
+}
+
+void EnSkjneedle_Destroy(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = THIS;
+	Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+int func_80B01F6C(Actor* thisx){
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:61]:
```suggestion
s32 func_80B01F6C(EnSkjneedle* this){
```
Read above

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 5]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/EnSkjneedle_Update.s")
+void EnSkjneedle_Update(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = THIS;
+	GlobalContext* preserve = globalCtx; //workaround to store globalCtx in $s1 and not on the stack
+	this->unk_1E0++;
+	if (this->unk_1E2 != 0){
+		this->unk_1E2--;
+	}
+	if (func_80B01F6C(&this->actor) || this->unk_1E2 == 0){
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:77]:
```suggestion
	if (func_80B01F6C(this) || this->unk_1E2 == 0){
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 6]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
     (ActorFunc)EnSkjneedle_Update,
     (ActorFunc)EnSkjneedle_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/EnSkjneedle_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/EnSkjneedle_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/func_80B01F6C.s")
+static ColliderCylinderInit_Set3 cylinderInit =
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:34]:
```suggestion
static ColliderCylinderInit_Set3 sCylinderInit = 
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 7]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
+static InitChainEntry initChain[] = {
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
+
+void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = (EnSkjneedle*)thisx;
+	Actor_ProcessInitChain(&this->actor, initChain);
+	Collider_InitCylinder(globalCtx, &this->collider);
+	Collider_SetCylinder_Set3(globalCtx, &this->collider, &this->actor, &cylinderInit);
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:50]:
```suggestion
	Collider_SetCylinder_Set3(globalCtx, &this->collider, &this->actor, &sCylinderInit);
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 8]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/func_80B01F6C.s")
+static ColliderCylinderInit_Set3 cylinderInit = 
+{
+	{ COLTYPE_UNK1, 0x11, 0x09, 0x00, COLSHAPE_CYLINDER },
+	{ 0x00, { 0xFFCFFFFF, 0x00, 0x08, }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x01, 0x01, 0x01 },
+	{ 10, 4, -2, { 0, 0, 0 } },
+};
+
+static InitChainEntry initChain[] = {
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:41]:
```suggestion
static InitChainEntry sInitChain[] = {
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 9]
```diff
@@ -27,13 +29,71 @@ const ActorInit En_Skjneedle_InitVars = {
...
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
+
+void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = (EnSkjneedle*)thisx;
+	Actor_ProcessInitChain(&this->actor, initChain);
```
Comment (shawlucas) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:48]:
```suggestion
	Actor_ProcessInitChain(&this->actor, sInitChain);
```

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 10]
```diff
@@ -14,8 +14,10 @@ void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnSkjneedle_Draw(Actor* thisx, GlobalContext* globalCtx);
+s32 func_80B01F6C(EnSkjneedle* this);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:17]: we generally like to keep a line break between the 4 main functions and the actor specific functions

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 11]
```diff
@@ -27,13 +29,70 @@ const ActorInit En_Skjneedle_InitVars = {
...
+};
+
+static InitChainEntry sInitChain[] = {
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
+
+void EnSkjneedle_Init(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = THIS;
+	Actor_ProcessInitChain(&this->actor, sInitChain);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:48]: another small thing, we like to keep a line break between stack variables and the code. this goes for the other functions in here too

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 12]
```diff
@@ -27,13 +29,70 @@ const ActorInit En_Skjneedle_InitVars = {
...
+	Actor_SetScale(&this->actor, 0.01f);
+}
+
+void EnSkjneedle_Destroy(Actor* thisx, GlobalContext* globalCtx){
+	EnSkjneedle* this = THIS;
+	Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80B01F6C(EnSkjneedle* this){
+	if ((this->collider.base.atFlags & 2) != 0){
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:62]: this is a flag check so you dont need != 0 here

File: src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c [hunk 13]
```diff
@@ -27,13 +29,70 @@ const ActorInit En_Skjneedle_InitVars = {
...
+{
+	{ COLTYPE_UNK1, 0x11, 0x09, 0x00, COLSHAPE_CYLINDER },
+	{ 0x00, { 0xFFCFFFFF, 0x00, 0x08, }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x01, 0x01, 0x01 },
+	{ 10, 4, -2, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c:44]: to keep clang from messing up formatting you should leave a trailing comma on the last entry. this also means you should run clang. youll need to install clang-format and clang-tidy then run format.sh which is in the root directory

**Pull request merged**